dist/PreserveModuleNamePlugin.js

```diff
@@ -73,6 +73,29 @@ function getPreservedModules(modules) {
                 m[exports.preserveModuleName] = req;
                 return true;
             }
+            // Since its relative, link it with the module alias if possible
+            const reasonRequest = r.module && r.module.rawRequest;
+            const moduleDependency = r.dependency && r.dependency.module;
+            const moduleRequest = moduleDependency && removeLoaders(moduleDependency.request);
+            if (req && aliases && reasonRequest && moduleRequest) {
+                const aliasMatches = Object.keys(aliases).filter((alias) => {
+                    const aliasPath = aliases[alias];
+                    const matchPath = !!(moduleRequest.match(new RegExp(`^${aliasPath}`)));
+                    const matchKey = !!(reasonRequest.match(new RegExp(`^${alias}`)));
+                    // We want to match against the alias file path AND the path that is being requested
+                    // This allows us to use relative files inside aliased files
+                    return (matchPath && matchKey);
+                });
+                // Invalid?
+                if (aliasMatches.length > 1) {
+                    throw new Error(`Incorrect alias usage. "${reasonRequest}" is duplicated in ${aliasMatches}`);
```

Comment:
So you're breaking the compilation if more than one alias could have produced a path? 
That seems wrong. For instance, I should be able to define all the aliases I want, as long as I don't use them there's no reason that the compilation should fail.